editor/plugins/tiles/tiles_editor_plugin.cpp

```diff
@@ -324,62 +324,117 @@ void TileMapEditorPlugin::_tile_map_changed() {
 }
 
 void TileMapEditorPlugin::_update_tile_map() {
-	TileMap *tile_map = Object::cast_to<TileMap>(ObjectDB::get_instance(tile_map_id));
-	if (tile_map) {
-		Ref<TileSet> tile_set = tile_map->get_tileset();
-		if (tile_set.is_valid() && edited_tileset != tile_set->get_instance_id()) {
-			tile_set_plugin_singleton->edit(tile_map->get_tileset().ptr());
+	TileMapLayer *edited_layer = Object::cast_to<TileMapLayer>(ObjectDB::get_instance(tile_map_layer_id));
+	if (edited_layer) {
+		Ref<TileSet> tile_set = edited_layer->get_effective_tile_set();
+		if (tile_set.is_valid() && tile_set_id != tile_set->get_instance_id()) {
+			tile_set_plugin_singleton->edit(tile_set.ptr());
 			tile_set_plugin_singleton->make_visible(true);
-			edited_tileset = tile_set->get_instance_id();
+			tile_set_id = tile_set->get_instance_id();
 		} else if (tile_set.is_null()) {
 			tile_set_plugin_singleton->edit(nullptr);
 			tile_set_plugin_singleton->make_visible(false);
-			edited_tileset = ObjectID();
+			tile_set_id = ObjectID();
 		}
 	}
 	tile_map_changed_needs_update = false;
 }
 
+void TileMapEditorPlugin::_select_layer(const StringName &p_name) {
+	TileMapLayer *edited_layer = Object::cast_to<TileMapLayer>(ObjectDB::get_instance(tile_map_layer_id));
+	ERR_FAIL_NULL(edited_layer);
+
+	Node *parent = edited_layer->get_parent();
+	ERR_FAIL_NULL(parent);
+
+	TileMapLayer *new_layer = Object::cast_to<TileMapLayer>(parent->get_node_or_null(String(p_name)));
+	edit(new_layer);
+}
+
+void TileMapEditorPlugin::_edit_tile_map_layer(TileMapLayer *p_tile_map_layer) {
+	ERR_FAIL_COND(!p_tile_map_layer);
+
+	editor->edit(p_tile_map_layer);
+
+	// Update the selected layers in the TileMapGroup parent node.
+	TileMapGroup *tile_map_group = Object::cast_to<TileMapGroup>(p_tile_map_layer->get_parent());
+	if (tile_map_group) {
+		Vector<StringName> selected;
+		selected.push_back(p_tile_map_layer->get_name());
+		tile_map_group->set_selected_layers(selected);
+	}
+
+	// Update the object ID.
+	tile_map_layer_id = p_tile_map_layer->get_instance_id();
+	p_tile_map_layer->connect("changed", callable_mp(this, &TileMapEditorPlugin::_tile_map_changed));
+
+	// Update the edited tileset.
+	Ref<TileSet> tile_set = p_tile_map_layer->get_effective_tile_set();
+	if (tile_set.is_valid()) {
+		tile_set_plugin_singleton->edit(tile_set.ptr());
+		tile_set_plugin_singleton->make_visible(true);
+		tile_set_id = tile_set->get_instance_id();
+	} else {
+		tile_set_plugin_singleton->edit(nullptr);
+		tile_set_plugin_singleton->make_visible(false);
+	}
+}
+
+void TileMapEditorPlugin::_edit_tile_map_group(TileMapGroup *p_tile_map_group) {
+	ERR_FAIL_COND(!p_tile_map_group);
+
+	Vector<StringName> selected_layers = p_tile_map_group->get_selected_layers();
+
+	TileMapLayer *selected_layer = nullptr;
+	if (selected_layers.size() >= 1) {
```

Comment:
```suggestion
	if (selected_layers.size() > 0) {
```

Reply:
I mean it's up to you but `>= 1` feels kinda weird, for an integer it's the same and it's less clear than `> 0`

Reply:
I don't know, it's the same to me... 

Anyway, it's changed on my local fork.

Reply:
> I don't know, it's the same to me...
> 
> Anyway, it's changed on my local fork.

I agree with your opinion. His changes are meaningless, and this is his meaningless speculative behavior

Reply:
@LiBooks Please refrain from personally attacking contributors like this. This is a clear breach of our Code of Conduct: https://godotengine.org/code-of-conduct/

I'll remove your post. Next occurrence of such inappropriate behavior would result in a ban.

Reply:
> I agree with your opinion. His changes are meaningless, and this is his meaningless speculative behavior

Well, thanks, but please avoid reopening that kind of threads on such minor details. It's not super constructive. (no harm done, just an advice for the future :) )
Thanks for having a look to the PR anyway!